flask_app/FlaskApp.py

```diff
@@ -1,16 +1,152 @@
+
+        await upload_video_task
+        thread_pool.shutdown(wait=True)
+
+        Logger.logger.info(f'finish to upload video for path: {path}')
+        Logger.logger.info(f'finish to upload images for path: {path}')
+        Logger.logger.info(f'Finish to insert video and frames to database, video path: {path}')
+        Logger.logger.info(f'Finish to handle path: {path}')
+
+        return json.dumps({'success': True}), 200, {'ContentType': 'application/json'}
+
+    def upload_images_to_azure(self, images: list, video_file_name: str, thread_pool: ThreadPoolExecutor):
+        """
+        Upload images to azure in background with thread pool
+        :param images: List of images
+        :param video_file_name: Video file name
+        :param thread_pool: Thread pool
+        """
+        for index, image in enumerate(images):
+            _, img_encode = cv2.imencode('.jpg', image)
+            img_bytes = img_encode.tobytes()
+            thread_pool.submit(self.azure_handler.upload_file, f'frame{index}.jpg',
+                               img_bytes, os.path.join('frames', video_file_name))
+
+    def create_and_insert_to_database_video_metadata_frame_models(self,
+                                                                  path: str,
+                                                                  image_list: list,
+                                                                  video_file_name: str):
+        """
+        Create and insert to database video metadata and frame models.
+        :param path: Video path.
+        :param image_list: Image list.
+        :param video_file_name: Video file name.
+        """
+        observation = self._get_observation_name_from_path(path)
+        video_from_db = self.db.get_video(observation, video_file_name, len(image_list))
+
+        if not video_from_db:
+            self.db.insert_one(data_models.Video(observation_name=observation,
+                                                 os_path=video_file_name,
+                                                 number_of_frames=len(image_list)))
+
+            video_model = self.db.get_video(observation_name=observation,
+                                            os_path=video_file_name,
+                                            number_of_frames=len(image_list))
+
+            frame_model_list = frame_utils.create_frame_models(len(image_list), video_model.id, video_file_name)
+
+            self.create_and_insert_metadata_and_set_metadata_id_to_frames(frame_model_list, image_list)
+
+            self.db.insert_many(frame_model_list)
+
+    def _get_observation_name_from_path(self, path: str):
+        """
+        Extract observation name from path.
+        :param path: Video file path.
+        :return: Observation name.
+        """
+        return os.path.basename(path).split('_')[0]
+
+    def create_and_insert_metadata_and_set_metadata_id_to_frames(self, frames: list, images: list):
```

Comment:
why having this function here and not in a new script that is designed for inserting data ?